### ISSUE TITLE
FilterLogger should respect inner logger's IsEnabled

### ISSUE THREAD

**Issue by andriysavin** [aspnet/Logging#534]
Currently, at [https://github.com/aspnet/Logging/blob/dev/src/Microsoft.Extensions.Logging.Filter/Internal/FilterLogger.cs#L27](https://github.com/aspnet/Logging/blob/dev/src/Microsoft.Extensions.Logging.Filter/Internal/FilterLogger.cs#L27) only provided filter is respected. But this functionality will likely be used for integration with existing logging frameworks, which may have their own filtering system. Why not respect it at least like this:
` 
public bool IsEnabled(LogLevel logLevel)
{
    return _filter(logLevel) && _innerLogger.IsEnabled(logLevel);
}
`

In other words, this filtering decorator should augment inner logger's filtering, not replace it.

**Comment by HaoK** [comment]
@pakrym was this fixed by your changes?

**Comment by pakrym** [comment]
Should be fixed, but I haven't verified.

**Comment by muratg** [comment]
Assigning to @pakrym for investigation.

**Comment by pakrym** [comment]
`Microsoft.Extensions.Logging.Filter` is getting removed. New way of adding filters has this fixed.
